src/config.rs

```diff
@@ -168,16 +177,31 @@ pub fn get_or_create(config_path: Option<&str>) -> Result<Config, String> {
     }
 
     match fs::File::open(&path) {
-        Ok(_) => Config::load(path)?
-            .check_for_timezone()?
-            .check_for_latest_version(),
+        Ok(_) => {
+            let config = Config::load(&path)?
+                .check_for_timezone()?
+                .check_for_latest_version()?;
+
+            // When we move the config file we also need to rename the path in JSON
+            if config.path != path {
+                let new_config = config.set_path(&path);
+                new_config.clone().save()?;
+                Ok(new_config)
+            } else {
+                Ok(config)
+            }
+        }
```

Comment:
I check to see if the path in the JSON differs from the path we looked it up with and edit/resave it if they are not the same.